### ISSUE TITLE
Update prompts: Neither "Install Update" nor "Skip This Version" buttons work

### ISSUE THREAD

**Issue by szhorvat** [RocketChat/Rocket.Chat.Electron#1871]
## My Setup

- Operating System: macOS 10.14.6
- App Version: 3.0.6 (note that updates can only be tested with the next-to-latest version!)
- Installation type: dmg from GitHub releases

- [ ] I have tested with the latest version **<-- I have tested with the next-to-latest version, as update prompts obviously don't appear in the the latest version**
- [x] I can simulate the issue easily

## Description

RocketChat 2.x on macOS has not updated correctly since at least 2.14. See #955 

RocketChat 3.0 made the situation worse in that clicking the "Skip this Version" button also does not work.

<img width="513" alt="image" src="https://user-images.githubusercontent.com/1212871/98671498-a658e200-2354-11eb-98ad-ec1f65ffddf2.png">

### Current Behavior

When clicking Install Update button:

 - "Update has been downloaded" prompt arrives, I click "Install Now"
 - The main RocketChat window disappears, but the app does not quit
 - Even after quitting manually, or using the "Install Later" button, the app does not update

When clicking the "Skip this Version" button:

 - If I quite the app and restart it, the same prompt comes back.

**tl;dr** Auto-update does not work (hasn't for years!!) and one cannot get rid of the update notifications either.

### Expected Behavior

 - Auto-update should work.
 - After clicking "Skip this Version", there should be no more update notifications until the next version becomes available

**Comment by Maqsyo** [comment]
Duplicate from #1837 #1670 #1205 and so on

**Comment by szhorvat** [comment]
@Maqsyo None of those mention "Skip this Update", which is newly broken.

**Comment by teak421** [comment]
I am also getting this odd behavior on the Windows client.  Enterprise customer, so I just opened a ticket up with support... Maybe just look and post here instead?

**Comment by teak421** [comment]
Downloaded 3.07 here and no issues to report.

**Comment by szhorvat** [comment]
> Downloaded 3.07 here and no issues to report.

Can you clarify? 3.0.7 is currently the latest version. It won't show an update box until 3.0.8 is released. Thus, currently, it is not possible to test this functionality with 3.0.7.

**Comment by teak421** [comment]
> > Downloaded 3.07 here and no issues to report.
> 
> Can you clarify? 3.0.7 is currently the latest version. It won't show an update box until 3.0.8 is released. Thus, currently, it is not possible to test this functionality with 3.0.7.

I had the exact same issue on 3.0.6... So, I updated manually.  But, you are correct, I should have been clearer.  Basically, I just upgraded manually, so I'm now on 3.0.7.  Let's hope that the issue doesn't repeat when 3.0.8 is out.

**Comment by AlbyGNinja** [comment]
I'm from 2.17.11 and confirm that update is not installed at all. Every time i click **"Install Update"** Rocket Chat start downloading the update into the path: `C:\Users\$user\AppData\Local\rocketchat-updater\pending` but right after download complete some kind of process delete it.
So basically Rocket Chat won't update and I can't even get the file manually.

(Windows 10 but this doesn't care really much)

**Comment by AlbyGNinja** [comment]
> I'm from 2.17.11 and confirm that update is not installed at all. Every time i click **"Install Update"** Rocket Chat start downloading the update into the path: `C:\Users\$user\AppData\Local\rocketchat-updater\pending` but right after download complete some kind of process delete it.
> So basically Rocket Chat won't update and I can't even get the file manually.
> 
> (Windows 10 but this doesn't care really much)

Ok i've workarounded it.
Just spam F5 into download folder above, when the icon appear the download is completed, then run it by yourself and voilà, installer is launched but no previous configurations are considered, so you have to login into servers you were logged before the update.

**Comment by ankar84** [comment]
> > I'm from 2.17.11 and confirm that update is not installed at all. Every time i click **"Install Update"** Rocket Chat start downloading the update into the path: `C:\Users\$user\AppData\Local\rocketchat-updater\pending` but right after download complete some kind of process delete it.
> > So basically Rocket Chat won't update and I can't even get the file manually.
> > 
> > (Windows 10 but this doesn't care really much)
> 
> Ok i've workarounded it.
> Just spam F5 into download folder above, when the file's icon appear the download is completed, then run it by yourself immediately and voilà, installer is launched **_but no previous configurations are considered_**, so you have to login into servers you were logged before the update.

Just download installer from release page is much easy.

**Comment by teak421** [comment]
> > > I'm from 2.17.11 and confirm that update is not installed at all. Every time i click **"Install Update"** Rocket Chat start downloading the update into the path: `C:\Users\$user\AppData\Local\rocketchat-updater\pending` but right after download complete some kind of process delete it.
> > > So basically Rocket Chat won't update and I can't even get the file manually.
> > > (Windows 10 but this doesn't care really much)
> > 
> > 
> > Ok i've workarounded it.
> > Just spam F5 into download folder above, when the file's icon appear the download is completed, then run it by yourself immediately and voilà, installer is launched **_but no previous configurations are considered_**, so you have to login into servers you were logged before the update.
> 
> Just download installer from release page is much easy.

For you, sure.  But, I'm in an enterprise environment wanting to use this product on hundreds of workstations.  It's stuff like this that shows the dev teams are not prioritizing correctly.  This should be fixed, ASAP.  How this got out of QA is beyond me.  Is there even an official QA process?

**Comment by ankar84** [comment]
> I'm in an enterprise environment

In that case just install it by enterprise solution like SCCM or InTune and disable autoupdate function

**Comment by teak421** [comment]
> > I'm in an enterprise environment
> 
> In that case just install it by enterprise solution like SCCM or InTune and disable autoupdate function

lol.  You explain to end-users about a stupid pop-up that comes up every time the application opens.  Help desk nightmare...

**Comment by ankar84** [comment]
> lol. You explain to end-users about a stupid pop-up that comes up every time the application opens. Help desk nightmare...

No, we turned off auto update functionality of Rocket Chat desktop application and deploy it using SCCM, so no one gets pop-up messages about new version, because that feature is disabled.

**Comment by teak421** [comment]
I get it... You like this software... I like this software... But, this software must perform well in an enterprise environment...not with workarounds. I do not have these issues with more polished software.  I WANT this software to succeed because it's AWESOME.  I just wish it focused on quality control more...prioritize fixes for the enterprise (like LDAP imports).  And, finally... have more responsive support.  All these are hallmarks of software that has been elevated to a higher level...past an open-source project.

I paid a hefty amount...and I am glad we did because I want to support this endeavor.  Just wish it had a tighter focus, that's all.

**Comment by ankar84** [comment]
> Wouldn't it be cool to have an auto-update that would work so you wouldn't have to package?

Actually auto-update feature [works fine in a pre Windows 10 1809 versions](https://github.com/RocketChat/Rocket.Chat.Electron/issues/1670#issuecomment-725355062)
I made an [issue about that](https://github.com/electron-userland/electron-builder/issues/5415)
 So, yes. I think that auto-update feature is a must have for such great software as Rocket Chat.



> I'm suggesting that focusing on enterprise features/fixes might make it help reach larger audiences.

Agree with you.

**Comment by ankar84** [comment]
Closing in favor of #1670
